slang-api/src/main/java/com/sonarsource/slang/visitors/TreeVisitor.java

```diff
@@ -34,14 +34,12 @@ public TreeVisitor() {
   }
 
   public void scan(C ctx, @Nullable Tree root) {
-    if (root != null) {
-      ctx.before(root);
-      before(ctx, root);
-      visit(ctx, root);
-      after(ctx, root);
-    }
+    ctx.before(root);
+    before(ctx, root);
+    visit(ctx, root);
+    after(ctx, root);
```

Comment:
How is this change related to integration tests?

Reply:
This was related to the empty_file.kt which resulted in a parsing error and had no metrics.
Reverted these changes in 1f44432452b84a70f79748bd21019f97ee7e0e85, as discussed, we consider it normal not to have measures for files with parsing errors